### ISSUE TITLE
Base images for cluster nodes

### ISSUE THREAD

**Issue by agradouski** [kubernetes-sigs/cluster-api-provider-kubevirt#89]
From CAPK community meeting discussion on Feb 1 (https://docs.google.com/document/d/1ZAnRLCKOVbDqrsrYis2OR0aZIAlqp576gCJVCkMNiHM), we need to find (or build and publish) a cluster node image, in order to truly unlock CAPK for developers. Also, it is a blocker for CAPK quick start guide.

Discussion points:
* currently, we use `kubevirtci` fedora image for testing: https://github.com/kubevirt/kubevirtci/blob/main/cluster-provision/images/vm-image-builder/fedora-kubeadm/cloud-config
* do we need to support multiple Linux versions? (so far, we have agreed that one is sufficient, e.g. ubuntu)
* image needs to have `kubeadm` installed
* a good example image should really be provided and possibly shipped with the release
* hopefully, `kubernetes-sigs/image-builder` project can save us some effort
* ideally, we can just use the openstack image of image-builder project: https://image-builder.sigs.k8s.io/capi/providers/openstack.html

**Comment by nirarg** [comment]
This is related to this issue #84
@isaacdorfman can you also please start investigating `kubernetes-sigs/image-builder` project and see if that's something we can use to generate disk images that work for capk

**Comment by wangxin311** [comment]
One comment: for CNI, we should leave it to user to choose just like other cloud provider doing for cluster api, end-user may not want or like built-in CNI provided us.

Here some of my experience on `image-builder`, FYI.
I built a preliminary image based on `kubernetes-sigs/image-builder` for openstack cluster api months ago, it does generate qcow2 image and it should work for kubevirt because both are based on KVM underneath, I gave a try on kubevirt cluster api, seems it working, not well intensively tested but at least workload kubernetes cluster is up, however some parameter may need to tune for kubevirt cluster api user case.
